src/FirebirdSql.Data.FirebirdClient/Common/Charset.cs

```diff
@@ -101,15 +106,17 @@ private static void TryAddCharset(List<Charset> charsets, Func<Charset> charsetC
 	public Encoding Encoding { get; }
 	public bool IsOctetsCharset { get; }
 	public bool IsNoneCharset { get; }
+	public FbCharsetEnum? FbCharset { get; }
 
-	public Charset(int id, string name, int bytesPerCharacter, string systemName)
+	public Charset(int id, string name, int bytesPerCharacter, string systemName, FbCharsetEnum? fbCharset = null)
```

Comment:
I think we need to think more about this. `FirebirdSql.Data.FirebirdClient.FbCharset` is basically use facing object while `Charset` is internal implementation detail. Either this needs to be split more or brought closer.